Apps/Database/src/Delegates/DbDelegationDelegate.cs

```diff
@@ -0,0 +1,168 @@
+{
+    using System.Collections.Generic;
+    using System.Linq;
+    using HealthGateway.Database.Constants;
+    using HealthGateway.Database.Context;
+    using HealthGateway.Database.Models;
+    using HealthGateway.Database.Wrapper;
+    using Microsoft.EntityFrameworkCore;
+    using Microsoft.Extensions.Logging;
+
+    /// <inheritdoc/>
+    public class DbDelegationDelegate : IDelegationDelegate
+    {
+        private readonly ILogger logger;
+        private readonly GatewayDbContext dbContext;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="DbDelegationDelegate"/> class.
+        /// </summary>
+        /// <param name="logger">The injected logger provider.</param>
+        /// <param name="dbContext">The context to be used when accessing the database.</param>
+        public DbDelegationDelegate(ILogger<DbDelegationDelegate> logger, GatewayDbContext dbContext)
+        {
+            this.logger = logger;
+            this.dbContext = dbContext;
+        }
+
+        /// <inheritdoc/>
+        public DbResult<AllowedDelegation> DeleteAllowedDelegation(AllowedDelegation allowedDelegation, bool commit = true)
+        {
+            DbResult<AllowedDelegation> result = new()
+            {
+                Payload = allowedDelegation,
+                Status = DbStatusCode.Deferred,
+            };
+            this.dbContext.AllowedDelegation.Remove(allowedDelegation);
+
+            if (commit)
+            {
+                try
+                {
+                    this.dbContext.SaveChanges();
+                    result.Status = DbStatusCode.Deleted;
+                }
+                catch (DbUpdateException e)
+                {
+                    result.Status = DbStatusCode.Error;
+                    result.Message = e.Message;
+                }
+            }
+
+            return result;
+        }
+
+        /// <inheritdoc/>
+        public DbResult<IList<Dependent>> GetDependents(string hdid, bool includeAllowedDelegation = false)
+        {
+            this.logger.LogTrace("Getting dependents - includeAllowedDelegation : {IncludeAllowedDelegation}", includeAllowedDelegation);
+            if (includeAllowedDelegation)
+            {
+                return new()
+                {
+                    Payload = this.dbContext.Dependent
+                        .Where(d => d.HdId == hdid)
+                        .Include(d => d.AllowedDelegations)
+                        .OrderByDescending(d => d.CreatedDateTime)
+                        .ToList(),
+                    Status = DbStatusCode.Read,
+                };
+            }
+
+            return new()
+            {
+                Payload = this.dbContext.Dependent
+                    .Where(d => d.HdId == hdid)
+                    .OrderByDescending(d => d.CreatedDateTime)
+                    .ToList(),
+                Status = DbStatusCode.Read,
+            };
+        }
```

Comment:
The query could be built bit-by-bit before executing to avoid repetition in the branching paths.

```
IQueryable<Dependent> query = this.dbContext.Dependent
    .Where(d => d.HdId == hdid);

if (includeAllowedDelegation)
{
    query = query.Include(d => d.AllowedDelegations);
}

return new()
{
    Payload = query.OrderByDescending(d => d.CreatedDateTime).ToList(),
    Status = DbStatusCode.Read,
};
```

Reply:
Modified as per suggestion.